acnportal/acnsim/events/stochastic_events.py

```diff
@@ -0,0 +1,277 @@
+from typing import List, Dict, Any
+from acnportal.acnsim.events import EventQueue, PluginEvent
+from acnportal.acnsim.models import EV, Battery
+import numpy as np
+from sklearn.mixture import GaussianMixture
```

Comment:
I added this as an import error, do you think that is good, or should we instead use a log message and exit to avoid the stack trace?

Reply:
Hmm, now that scikit learn is in setup.py, I don't think we really need an error at all actually. It's on the same footing as numpy and pandas and we don't throw errors for those.

Reply:
Sort of, while numpy and pandas are required by default scikit-learn is only installed if you install the extras, so maybe still helpful to have the message? I don't want to install it by default since its kinda a big package.

Reply:
Nevermind, I saw that you added it as an extra. The error looks good. Since it would anyway throw an error if scikit learn was missing, I don't think logging it instead is that helpful.